api/envoy/config/listener/v3/listener.proto

```diff
@@ -93,6 +95,23 @@ message Listener {
   message InternalListenerConfig {
   }
 
+  // Filter chain with a name.
+  message NamedFilterChain {
+    // Filter chain name.
+    string name = 1 [(validate.rules).string = {min_len: 1}];
+
+    oneof config_type {
+      // Filter chain configuration.
+      FilterChain filter_chain = 2;
+
+      // Configuration source specifier for an extension configuration discovery
+      // service. In case of a failure and without the default configuration, the
+      // connection is rejected.
```

Comment:
Can you make it clear this is optional.

Reply:
This is not-implemented-hide, and I think can wait for later. I want this for completeness.